### PR TITLE
fix: modify the icon type to optional and condition the props request

### PR DESCRIPTION
**Pull request by CodeLeom**

### CODE REVIEW

File: src/components/element/Button.astro [hunk 1]
```diff
@@ -4,7 +4,7 @@ interface Props {
   label: string;
   style: "primary" | "secondary";
   size: "md" | "lg";
-  icon: any; //for a button with an icon
+  icon?: string; //for any button with an icon
```
Comment (codejagaban) [inline src/components/element/Button.astro:7]: Import 

```
import type { ImageMetadata } from "astro";
```
and use `ImageMetadata` here

File: src/components/element/Button.astro [hunk 2]
```diff
@@ -22,5 +22,5 @@ const { href, label, style, size, icon } = Astro.props;
     },
   ]}
 >
-  {(<img src={icon} alt="Icon" class="inline-block mr-2" />)} {label}</a
+  {icon && <img src={icon} alt="Icon" class="inline-block mr-2" />} {label}</a
```
Comment (codejagaban) [inline src/components/element/Button.astro]: the alt should be the button label

File: src/pages/404.astro [hunk 3]
```diff
@@ -50,6 +42,12 @@ import Button from "../components/element/Button.astro"; //button component
           />
         </div>
       </div>
+
+      <!-- Image -->
+      <div class="image-container w-full sm:order-1">
+        <img src="/error.png" class="mb-0 md:mb-10 animated-image" alt="error image" />
```
Comment (codejagaban) [inline src/pages/404.astro:39]: use the `Image` component from Astro to optimize this image

**Review by codejagaban — CHANGES_REQUESTED** [review]
@CodeLeom the build is failing, can you please merge `develop` into your branch so we can test this?

File: src/components/element/Button.astro (hunk 1 above)
Comment (CodeLeom) [reply src/components/element/Button.astro:7]: Screenshot 2023-12-30 at 19 24 14
Is this the expected usage? on line 9?

Comment (codejagaban) [reply src/components/element/Button.astro:7]: Yes this works

Comment (codejagaban) [reply src/components/element/Button.astro:7]:
```
interface Props {
...
icon?: ImageMetadata;


}
```

File: public/fi_home.png
Comment (codejagaban) [inline public/fi_home.png:1]: Please import these as SVG files so they can maintain their quality

File: public/mail.png
Comment (codejagaban) [inline public/mail.png:1]: SVG for this also

**Review by codejagaban — CHANGES_REQUESTED** [review]
@CodeLeom Please update the tiny changes I requested, I'll fix the others when we merge it.

**Comment by codejagaban** [comment]
Also, @CodeLeom, the page is not responsive; please fix that 🤲🏼

**Comment by codejagaban** [comment]
I will be closing This in favor of #23 🫡

**Pull request closed without merging**